newproject.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/sh
+set -e
+
+# Script inspired by https://gist.github.com/szeidner/613fe4652fc86f083cefa21879d5522b
+
+PROGNAME=$(basename $0)
+WORKING_DIR=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+
+die() {
+    echo "$PROGNAME: $*" >&2
+    exit 1
+}
+
+usage() {
+    if [ "$*" != "" ] ; then
+        echo "Error: $*"
+    fi
+
+    cat << EOF
+Usage: $PROGNAME --template [TEMPLATE] --package-name [PACKAGE_NAME] --name [APP_NAME]
+Set up an Android app and package name.
+Options:
+-h, --help                         display this usage message and exit
+-t, --template [TEMPLATE]          Select template: "rx" - RxJavaTemplate or "crt" - CoroutineTemplate (i.e. rx)
+-p, --package-name [PACKAGE_NAME]  new package name (i.e. com.example.package)
+-n, --app-name [APP_NAME]          new app name (i.e. MyApp, "My App")
+EOF
+
+    exit 1
+}
+
+template=""
+packagename=""
+appname=""
+while [ $# -gt 0 ] ; do
+    case "$1" in
+    -h|--help)
+        usage
+        ;;
+    -t|--template)
+        template="$2"
+        shift
+        ;;
+    -p|--package-name)
+        packagename="$2"
+        shift
+        ;;
+    -n|--app-name)
+        appname="$2"
+        shift
+        ;;
+    -*)
+        usage "Unknown option '$1'"
+        ;;
+    *)
+        usage "Too many arguments"
+      ;;
+    esac
+    shift
+done
+
+OLD_APPNAME=""
+OLD_NAME=""
+OLD_PACKAGE=""
+
+# Path segments
+FIRST_PACKAGE_SEGMENT="co"
+SECOND_PACKAGE_SEGMENT="nimblehq"
+THIRD_PACKAGE_SEGMENT=""
+# Select template
+case "$template" in
+    "crt")
+        OLD_APPNAME="Coroutine Template"
+        OLD_NAME="CoroutineTemplate"
+        OLD_PACKAGE="co.nimblehq.coroutine"
+        THIRD_PACKAGE_SEGMENT="coroutine"
+        ;;
+    "rx")
+        OLD_APPNAME="RxJava Template"
+        OLD_NAME="RxJavaTemplate"
+        OLD_PACKAGE="co.nimblehq.rxjava"
+        THIRD_PACKAGE_SEGMENT="rxjava"
+        ;;
+    -*)
+        usage "Unknown template '$template'. Please use 'rx' for RxJavaTemplate or 'crt' for CoroutineTemplate"
+        ;;
+esac
+
+if [ -z "$packagename" ] ; then
+    usage "No new package provided"
+fi
+
+if [ -z "$appname" ] ; then
+    usage "No new app name provided"
+fi
+
+# Enforce package name
+regex='^[a-z][a-z0-9_]*(\.[a-z0-9_]+)+[0-9a-z_]$'
+if ! [[ $packagename =~ $regex ]]; then
+    die "Invalid Package Name: $packagename (needs to follow standard pattern {com.example.package})"
+fi
+
+echo "=> 🐢 Staring init $appname with $OLD_APPNAME..."
+
+# Trim spaces in APP_NAME
+NAME_NO_SPACES=$(echo "$appname" | sed "s/ //g")
+
+# Copy main folder
+cp -R $OLD_NAME $NAME_NO_SPACES
+
+# get rid of idea settings
+rm -rf $NAME_NO_SPACES/.idea
```

Comment:
same as here, as we already polish our template repo to not commit this directory as well